### PR TITLE
Remove command palette POC

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Remove a POC for the Command Palette feature, no longer on the roadmap. APIs behind a feature flag are considered unsupported, and thus removed without addressing semantic versioning. The schema changes remain intact, but with a comment to explain their non-use. And finally a set of improvements that improved the existing code is left intact, but some changes to the semantics of `team_id` remain reverted -- see comments inline.

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-53349

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -1547,16 +1547,14 @@ func (s SqlChannelStore) getByName(teamId string, name string, includeDeleted bo
...
-	if !includeDeleted {
-		query = query.Where(sq.Eq{"DeleteAt": 0})
-	}
-	if teamId != "" {
-		query = query.Where(sq.Or{
+		Where(sq.Eq{"Name": name}).
+		Where(sq.Or{
 			sq.Eq{"TeamId": teamId},
 			sq.Eq{"TeamId": ""},
 		})
```
Comment (lieut-data) [inline server/channels/store/sqlstore/channel_store.go:1554]: The original [POC](https://github.com/mattermost/mattermost/commit/aa59c28b04fa45f0b02e5149ac0fbcd2b78420b9) changed the semantics here to treat `teamId != ""` as allowing any team, but in at least one case we assumed `teamId == ""` meant only direct channels:

https://github.com/mattermost/mattermost/blob/823ef27d6c5351925de667da243df4bf458ce07b/server/channels/app/channel.go#L3491-L3492

To keep things consistent without a clear reason to keep the wider search, I reverted back to the old semantics for `C.TeamId = :teamId OR c.TeamId = ""` everywhere, while preserving the SQL Builder changes where applicable.

File: server/channels/store/sqlstore/file_info_store.go [hunk 2]
```diff
@@ -512,33 +511,12 @@ func (fs SqlFileInfoStore) Search(paramsList []*model.SearchParams, userId, team
 		From("FileInfo").
 		LeftJoin("Channels as C ON C.Id=FileInfo.ChannelId").
 		LeftJoin("ChannelMembers as CM ON C.Id=CM.ChannelId").
+		Where(sq.Or{sq.Eq{"C.TeamId": teamId}, sq.Eq{"C.TeamId": ""}}).
```
Comment (lieut-data) [inline server/channels/store/sqlstore/file_info_store.go:514]: See comment above re: changes to `C.TeamId`.

File: server/channels/store/sqlstore/post_store.go [hunk 3]
```diff
@@ -1923,13 +1921,11 @@ func (s *SqlPostStore) buildSearchPostFilterClause(teamID string, fromUsers []st
...
-	sb := s.getSubQueryBuilder().
-		Select("Id").
-		From("Users, TeamMembers").
-		Where(sq.Expr("Users.Id = TeamMembers.UserId"))
-	if teamID != "" {
-		sb = sb.Where(sq.Eq{"TeamMembers.TeamId": teamID})
-	}
+	sb := s.getSubQueryBuilder().Select("Id").From("Users, TeamMembers").Where(
+		sq.And{
+			sq.Eq{"TeamMembers.TeamId": teamID},
```
Comment (lieut-data) [inline server/channels/store/sqlstore/post_store.go:1926]: See changes above re: `TeamId`.

File: server/public/model/search_params.go [hunk 4]
```diff
@@ -33,7 +33,7 @@ type SearchParams struct {
 	IncludeDeletedChannels bool     `json:"include_deleted_channels,omitempty"`
 	TimeZoneOffset         int      `json:"timezone_offset,omitempty"`
 	// True if this search doesn't originate from a "current user".
-	SearchWithoutUserId bool   `json:"search_without_userid,omitempty"`
+	SearchWithoutUserId bool   `json:"search_without_user_id,omitempty"`
```
Comment (lieut-data) [inline server/public/model/search_params.go:36]: Kept the `json` annotations, but fixed the key `..._user_id` to be consistent with other JSON fields.

**Review by agnivade — APPROVED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]

**Comment by lieut-data** [comment]
@lindalumitchell, I've run the e2e-tests and got a great passing coverage. I've also spot checked the functionality touched by this PR. Might you be able to recommend next steps for QA Review in this case?

**Review by lindalumitchell — APPROVED** [review]
Thanks @lieut-data! I looked through, and given you've already checked the e2es, I'm happy to approve this. 👍

**Pull request merged**